### DISCUSSION TITLE
Endless Loop of Emails - Hard Fail, Retry and Error

### DISCUSSION THREAD

**Discussion by ecalpemos2023** [postalserver/postal#2726] [Help with using Postal]
I have an endless loop of emails that are always in the queue, although the queue shows ZERO or Two, infact there are currently 56 emails in this state and that is after me manually removing over 200 from the queue.

![image](https://github.com/postalserver/postal/assets/154352780/a1abacdc-21be-4f6f-8eeb-9e656072c7cf)

The emails themselves seem to be bounce notifications from Mimecast, but there is never a Route or a Domain, they are stuck in an endless route of Hard Fail, Retry and Error.

![image](https://github.com/postalserver/postal/assets/154352780/25cd214a-3c10-4d8c-a245-e15507740d93)

![image](https://github.com/postalserver/postal/assets/154352780/03bbf9cd-944f-4fa6-b910-7c20c3e0dbad)

Somethings I read have suggested changing the 
Find line 51: output: output.to_s.force_encoding("UTF-8").scrub.truncate(512),
Change it to: output: output.to_s.dup.force_encoding("UTF-8").scrub.truncate(512),

I have done this in all containers, and I have set retry to 2 but it continues forever.
It may be a DNS setting that is incorrect but if the bounce is not Mimecast related, it works fine.

Does anyone have any other suggestions on what I could try? I am by no means a programmer, but willing to learn!

**Comment by willpower232** [comment]
If you click on "Show technical details" it might explain what is happening.

Does the content of the messages make sense to you? Are you expecting bounces or other notifications from them?

**Reply by ecalpemos2023** [reply]
Every message says the same: FrozenError: can't modify frozen String: ""
so I changed the Line 51 as the forum states etc but I dont understand why it continues to try and send if it is a Hard Fail and why it only occurs to Mimecast bounces, whereas bounces from other domains are processed just fine. It has also led to us having a 113% bounce rate as I think everytime it hard fails, it counts against bounce rate and IP reputation.

It is almost like Mimecast removes any reference to the original message and therefore Postal can not process the returned bounce message:
This message was a bounce but we couldn't link it with any outgoing message and there was no route for it.


So far my only solution is to manually delete all the emails from the queued messages, this seems to stop the loop.

**Reply by tb-bb** [reply]
We have the same problem and have no explanation whatsoever. Are there any new findings or ideas on this?

**Reply by Pacerino** [reply]
@gonzo686 Take a look into this script: https://github.com/postalserver/postal/issues/2153#issuecomment-1879626081

This resolved my problem with the `FrozenError: can't modify frozen String: ""` error
Edit: You probably need to rename the container name string in the script from `name=postal_" --format "{{.ID}}")` to `name=postal-" --format "{{.ID}}")` because my container were named like `postal-` and not `postal_`

**Reply by ecalpemos2023** [reply]
@gonzo686 I am by no means a programmer or developer and I just couldnt get the solutions to work. In the end I hiired a developer who chrged me $40 dollars to sort and to sort out all mt DNS settings etc. If you want their contact info you can msg me. They resolved all my issue sin a day.

**Reply by tb-bb** [reply]
Thank you for your quick responses! @Pacerino your mentioned comment solved our problems! Thank you!
